transports/transports.go

```diff
@@ -30,10 +26,10 @@ func init() {
 		storage.Transport,
 	} {
 		name := t.Name()
-		if _, ok := KnownTransports[name]; ok {
-			panic(fmt.Sprintf("Duplicate image transport name %s", name))
+		if _, ok := registeredtransports.KnownTransports[name]; ok {
+			continue
 		}
-		KnownTransports[name] = t
+		registeredtransports.Register(t)
```

Comment:
These transports should have been registered merely by the fact of being imported here, shouldn’t they? So this could just verify their presence and panic if they are not already in `KnownTransports`. Or is the import and initialization order uncertain?